faculty/clients/experiment.py

```diff
@@ -245,6 +350,113 @@ def make_restore_runs_response(self, data):
+
+    def _determine_field(self, value):
+        if isinstance(value, str):
+            return fields.String()
+        elif isinstance(value, int) or isinstance(value, float):
+            return fields.Number()
+        else:
+            self.fail("unsupported_type")
+
+    def _deserialize(self, value, attr, data, **kwargs):
+        field = self._determine_field(value)
+        return field._deserialize(value, attr, data, **kwargs)
+
+    def _serialize(self, value, attr, obj, **kwargs):
+        field = self._determine_field(value)
+        return field._serialize(value, attr, obj, **kwargs)
+
+
+class SingleFilterValueField(fields.Field):
+    """
+    Field that serialises/deserialises a run filter.
+    """
+
+    default_error_messages = {
+        "invalid_filter_operator": "Invalid filter operator."
+    }
+
+    FILTER_BY_FIELD_MAPPING = {
+        SingleFilterBy.PROJECT_ID: fields.UUID,
+        SingleFilterBy.RUN_ID: fields.UUID,
+        SingleFilterBy.EXPERIMENT_ID: fields.Integer,
+        SingleFilterBy.DELETED_AT: fields.DateTime,
+        SingleFilterBy.TAG: fields.String,
+        SingleFilterBy.PARAM: ParamValueField,
+        SingleFilterBy.METRIC: fields.Number,
+    }
+
+    def _deserialize(self, value, attr, data, **kwargs):
+        return value
+
+    def _serialize(self, value, attr, obj, **kwargs):
+        if obj.operator == SingleFilterOperator.DEFINED:
+            field_cls = fields.Boolean
+        else:
+            try:
+                field_cls = self.FILTER_BY_FIELD_MAPPING[obj.by]
+            except KeyError:
+                self.fail("invalid_filter_operator")
+        return field_cls()._serialize(value, attr, obj, **kwargs)
+
+
+class FilterField(fields.Field):
+    """
+    Field that serialises/deserialises a run filter.
+    """
+
+    default_error_messages = {"invalid_none_filter": "A none filter"}
+
+    def _deserialize(self, value, attr, data, **kwargs):
+        if value is None:
+            return None
+        elif isinstance(value, SingleFilter):
+            return SingleFilterSchema().load(value)
+        else:
+            return CompoundFilterSchema().load(value)
+
+    def _serialize(self, value, attr, obj, **kwargs):
+        if value is None and isinstance(obj, QueryRuns):
+            return None
+        elif value is None:
+            self.fail("invalid_none_filter")
+        if isinstance(value, SingleFilter):
+            return SingleFilterSchema().dump(value)
+        else:
+            return CompoundFilterSchema().dump(value)
+
+
+class SingleFilterSchema(BaseSchema):
+    by = EnumField(SingleFilterBy, by_value=True, required=True)
+    key = fields.String()
+    operator = EnumField(SingleFilterOperator, by_value=True, required=True)
+    value = SingleFilterValueField(required=True)
+
+
+class CompoundFilterSchema(BaseSchema):
+    operator = EnumField(CompoundFilterOperator, by_value=True, required=True)
+    conditions = fields.List(FilterField(skip_if=None))
```

Comment:
What does `skip_if` do? I can't find any mention of it in the documentation.

Reply:
I think this was added by mistake. Thanks for spotting this out, I will remove it now.